### DISCUSSION TITLE
Change themes without Theme Tool GUI

### DISCUSSION THREAD

**Discussion by MachoMan1988** [namazso/SecureUxTheme#156] [Ideas]
Is there a way to change themes without using the Theme Tool GUI? I'd like to switch themes by scripting with apps that offer the script function such as Auto Dark Mode, so switching by Light to Dark mode with patched themes would be easy by hooking SecureUxTheme functions.
Thanks for your attention.

**Comment by namazso** [comment]
Yes, this is the purpose of ThemeDll. See interface in https://github.com/namazso/SecureUxTheme/tree/master/public

**Reply by MachoMan1988** [reply]
so are there istructions to make it work as a script with cmd prompt so I could make some .bat/.ps1/.vbs scripts?

**Reply by namazso** [reply]
No, that's the part you're doing. You can use P/Invoke in PowerShell to call the functions.

**Reply by razor19995** [reply]
@namazso, any docs with the functions explained? Or just the names of the functions?
I assume that we have opportunity to install and change the themes via batch, but no-one knows how to?

**Reply by namazso** [reply]
@razor19995 the headers were just moved to https://github.com/namazso/SecureUxTheme/tree/master/ThemeLib/public

All of them come with complete doxygen-style docs.

**Comment by dreadbalaur** [comment]
Just wondering if you ever figured out how to hook the functions? I'm looking to do the same thing.
